zilliqa/src/consensus.rs

```diff
@@ -645,6 +645,7 @@ impl Consensus {
                         contract_address: result.contract_address,
                         logs: result.logs,
                         gas_used: result.gas_used,
+                        scilla_events: serde_json::to_string(&result.scilla_events).unwrap(),
```

Comment:
Unfortunetly, the TransactionReceipt needs to have a concrete type since it gets written down to disk. Using a serde_json::Value doesn't work as it doesn't know how much space to allocate it essentially. So it gets set to a string here. There might well be a way to get around this more cleanly

Reply:
I don't understand why `serde_json::Value` doesn't work here? It represents things internally with dynamic allocation so it shouldn't need to know how much space it allocates in advance.

Reply:
So I re-ran it, it will generate this panic at runtime:

```
2024-01-24T16:43:03.827129Z ERROR zilliqa::api::eth: e=Bincode does not support the serde::Deserializer::deserialize_any method

Stack backtrace:
   0: std::backtrace_rs::backtrace::libunwind::trace
             at /rustc/eeff92ad32c2627876112ccfe812e19d38494087/library/std/src/../../backtrace/src/backtrace/libunwind.rs:104:5
   1: std::backtrace_rs::backtrace::trace_unsynchronized
             at /rustc/eeff92ad32c2627876112ccfe812e19d38494087/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2: std::backtrace::Backtrace::create
             at /rustc/eeff92ad32c2627876112ccfe812e19d38494087/library/std/src/backtrace.rs:331:13
   3: std::backtrace::Backtrace::capture
             at /rustc/eeff92ad32c2627876112ccfe812e19d38494087/library/std/src/backtrace.rs:296:9
   4: anyhow::error::<impl core::convert::From<E> for anyhow::Error>::from
             at /Users/nhutton/.cargo/registry/src/index.crates.io-6f17d22bba15001f/anyhow-1.0.79/src/error.rs:565:25
   5: <core::result::Result<T,F> as core::ops::try_trait::FromResidual<core::result::Result<core::convert::Infallible,E>>>::from_residual
             at /rustc/eeff92ad32c2627876112ccfe812e19d38494087/library/core/src/result.rs:1959:27
   6: zilliqa::db::Db::get_transaction_receipts
             at ./zilliqa/src/db.rs:227:5
```

What I wrote before isn't actually correct, it is on the deserialization, it doesn't know what type to deser into, as mentioned here: https://stackoverflow.com/questions/77315666/how-can-i-deserialize-a-bincode-serialised-data-into-something-like-serde-json

So it seems like it is something like, one of bincode or the struct it is getting deseralized into need to know the type/size, but neither of them know in the case of `serde_json::Value`